templates/nginx.conf.j2

```diff
@@ -10,6 +10,36 @@ server {
 }
 {% endif -%}
 
+{% for redirector in app_nginx_redirect_names %}
+{% if redirector.ssl_enabled and redirector.ssl_only -%}
+server {
+    server_name {{ redirector.name }};
+    listen 80; listen [::]:80;
+    return {{ redirector.redirect_code }} https://{{ app_name }}$request_uri;
+}
+{% endif -%}
+
+server {
+    server_name {{ redirector.name }};
+
+    {% if not redirector.ssl_only %}
+    listen 80; listen [::]:80;
+    {% endif %}
+
+    {% if redirector.ssl_enabled -%}
+    listen 443 ssl {% if redirector.ssl_spdy_enabled %}http2{% endif %};
+    listen [::]:443 ssl {% if redirector.ssl_spdy_enabled %}http2{% endif %};
+
+    ssl_certificate {{ redirector.ssl_cert_path }};
+    ssl_certificate_key {{ redirector.ssl_priv_path }};
+
+    return {{ redirector.redirect_code }} https://{{ app_name }}$request_uri;
```

Comment:
@peakwinter Here I think that using `app_nginx_ssl_enabled` for the `return` would be better. For example :

```nginx
return {{ redirector.redirect_code }} http{% if app_nginx_ssl_enabled %}s{% endif %}://{{ app_name }}$request_uri;
```

Making this outside of the `{% if redirector.ssl_enabled -%}` statement.

The redirect instance can be SSL, but its redirect is not inherent to the application instance.

Do you think that this is a good idea ?

Reply:
Complete example :

```nginx

{% for redirector in app_nginx_redirect_names %}
server {
    server_name {{ redirector.name }};

    {% if not redirector.ssl_only %}
    listen 80; listen [::]:80;
    {% endif %}

    {% if redirector.ssl_enabled -%}
    listen 443 ssl {% if redirector.ssl_spdy_enabled %}http2{% endif %};
    listen [::]:443 ssl {% if redirector.ssl_spdy_enabled %}http2{% endif %};

    ssl_certificate {{ redirector.ssl_cert_path }};
    ssl_certificate_key {{ redirector.ssl_priv_path }};
    {% endif %}

    return {{ redirector.redirect_code }} http{% if app_nginx_ssl_enabled %}s{% endif %}://{{ app_name }}$request_uri;
}
{% endfor %}
```